Angular/Angular guidelines and best practices/File and module organization and naming.md

```diff
@@ -15,53 +15,113 @@ Try to keep the code grouped by domain. For example, if you have a `Post` class,
 src/app/services/post/
 ├── post.service.ts
 ├── post.service.spec.ts
+├── post.testing.service.ts
 ├── post.interface.ts
 ├── post.model.ts
 └── post.model.spec.ts
 ```
 
-## Pages
+## Pages and shared components
 
-Container components that are loaded directly via routing (by either `component` or `loadChildren`) should be placed in the `src/app/pages` directory.
+Components that are loaded directly via routing (by either `component` or `loadChildren`) should be placed in the `src/app/pages` directory. If a component is used in multiple places, it should be extracted to the nearest common ancestor's components directory. If there are multiple levels of routing, the main `src/app` directory structure can be mirrored under some specific page directory. In other words, you can have components, services, pages, etc. directories inside some specific page directory. This helps separate the codebase on a feature/domain-basis.
 
-## Shared components
+As an example, please consider the following routing setup:
 
-If a component is used in multiple places, it should be extracted to the nearest common ancestor's components directory. Here are some examples:
+- `/` - homepage, redirects to `/posts`
+- `/users` - list of users (renders a list of `user-avatar` components)
+- `/users/:id/view` - user details view (renders `user-avatar` and a list of `post-overview` components)
```

Comment:
Do we really want to force the "view" suffix here?

Reply:
I am flexible on this. With REST APIs we have different HTTP verbs so the routes are usually just `/users/:id` and then you have GET, POST, etc. I am not sure what's the most common thing for frontend apps, I've seen many different approaches (`users/:id`, `users/:id/view`, `users/:id/details`, `users/view/:id`, etc.)

Reply:
My view (hehe) on this is that there is no reason to specify we are viewing a single resource with additional suffix in the URL, :id of a resource is enough to clarify that, with additional verbs/nouns such as "edit" implying a different action a resource, in this case, one can understand /users/:id:/view as us browsing "a view of users :id", where "a view" != "a detail of a resource".

Reply:
Sold, I will drop `/view`